### ISSUE TITLE
How can I generate WMS requests with lowercase parameters

### ISSUE THREAD

**Issue by Amither113** [MapServer/mapcache#325]
MapCache defaults to uppercase parameters in WMS requests, is there a configuration method to make the request parameters in lowercase?

**Comment by jratike80** [comment]
Please clarify what you mean with parameters. By the WMS standard:

> 6.4 Request Parameter Rules 
> 6.4.1 Parameter Ordering and Case 
> Parameter names shall not be case sensitive, but parameter values shall be case sensitive. 
> In this document, parameter names are typically shown in uppercase for typographical 
> clarity, not as a requirement. 

It means that by the standard only "WMS" should be accepted as a service name, but `&SerVICe=WMS` is OK. Most, but not all, WMS servers accept also "wms".

Do you know some client that does not follow the standard?

**Comment by Amither113** [comment]
Yes, I have encountered WMS services that do not follow the standard, when I use `&service=wms` I can request successfully, but not when `&SERVICE=WMS`.

**Comment by jratike80** [comment]
The primary action would be to ask if they could make a fix on their side. It also fits with the robustness principle https://en.wikipedia.org/wiki/Robustness_principle.

**Comment by marisn** [comment]
There are infinitely many ways how a non-conformant server can behave and thus it is impossible to support non-conformant servers as infinitely many workarounds would have to be implemented. This is the reason why we do have standards and follow them.
